pyanaconda/payload/live/payload_liveos.py

```diff
@@ -41,20 +42,36 @@ def type(self):
         """The DBus type of the payload."""
         return PAYLOAD_TYPE_LIVE_OS
 
+    @staticmethod
+    def _get_live_os_image():
+        """Detect live os image on the system.
+
+        :return: a path to the image
+        """
+        payloads_proxy = PAYLOADS.get_proxy()
+        source_path = payloads_proxy.CreateSource(SOURCE_TYPE_LIVE_OS_IMAGE)
+
+        source_proxy = PAYLOADS.get_proxy(source_path)
+        return source_proxy.DetectLiveOSImage()
```

Comment:
This should be handled better way I think. It's fine for now but I don't think we should create source to be able to detect if there is a source. This should be ideally part of the payload?

Please add TODO here.

Reply:
Added a comment.